test/mpi/rma/putfence1.c

```diff
@@ -7,43 +7,55 @@
 #include "mpi.h"
 #include <stdio.h>
 #include <string.h>
+#include <limits.h>
 #include "mpitest.h"
-
-/* These counts allow reasonable sizes for the large tests */
-#define LARGE_CNT_CONTIG    550000000
-#define LARGE_CNT_NONCONTIG 150000000
+#include "dtpools.h"
 
 /*
 static char MTEST_Descrip[] = "Put with Fence";
 */
 
 static inline int test(MPI_Comm comm, int rank, int source, int dest,
-                       MTestDatatype * sendtype, MTestDatatype * recvtype)
+                       int i, int j, DTP_t send, DTP_t recv)
 {
     int errs = 0, err;
+    int len;
+    int disp_unit;
+    MPI_Aint sendcount, recvcount, count;
     MPI_Aint extent, lb;
     MPI_Win win;
+    MPI_Datatype sendtype, recvtype;
+    char send_name[128] = { 0 };
+    char recv_name[128] = { 0 };
+    void *sendbuf, *recvbuf;
+
+    count = send->DTP_type_signature.DTP_pool_basic.DTP_basic_type_count;
+    sendtype = send->DTP_obj_array[i].DTP_obj_type;
+    recvtype = recv->DTP_obj_array[j].DTP_obj_type;
+    sendcount = send->DTP_obj_array[i].DTP_obj_count;
+    recvcount = recv->DTP_obj_array[j].DTP_obj_count;
+    sendbuf = send->DTP_obj_array[i].DTP_obj_buf;
+    recvbuf = recv->DTP_obj_array[j].DTP_obj_buf;
+
+    MPI_Type_get_name(sendtype, send_name, &len);
+    MPI_Type_get_name(recvtype, recv_name, &len);
 
     MTestPrintfMsg(1,
                    "Putting count = %ld of sendtype %s - count = %ld receive type %s\n",
-                   sendtype->count, MTestGetDatatypeName(sendtype), recvtype->count,
-                   MTestGetDatatypeName(recvtype));
-
-    /* Make sure that everyone has a recv buffer */
-    recvtype->InitBuf(recvtype);
-    MPI_Type_extent(recvtype->datatype, &extent);
-    MPI_Type_lb(recvtype->datatype, &lb);
-    MPI_Win_create(recvtype->buf, recvtype->count * extent + lb, extent, MPI_INFO_NULL, comm, &win);
+                   sendcount, send_name, recvcount, recv_name);
+
+    MPI_Type_extent(recvtype, &extent);
+    MPI_Type_lb(recvtype, &lb);
+    disp_unit = extent < INT_MAX ? extent : 1;
+    MPI_Win_create(recvbuf, recvcount * extent + lb, disp_unit, MPI_INFO_NULL, comm, &win);
```

Comment:
Why `disp_unit = extent < INT_MAX ? extent : 1;` ?

Reply:
This was failing for large counts I had to add it